operator/cmd/mesh/testdata/manifest-generate/output/component_hub_tag.golden.yaml

```diff
@@ -8993,5 +8572,426 @@ webhooks:
 
 # Policy component is disabled.
 
+# Resources for Prometheus component
```

Comment:
That's sort of the point of this PR, why rollback on that? Just because the manifests are re-ordered?

The idea was that we don't have a fake "AddonComponent" that really is multiple components, because like that, we have no way of knowing what we're really applying later in the codepath, except by looking at the actual manifests (which is not really helpful).

Reply:
The only thing I'm pushing back on in this PR is reintroducing the addon component names back into the code - I'm agnostic to the implementation. It would defeat the purpose of having a generic API for addon components and then add them back into the code.
I agree AddonComponent is a bit weird (note that ingress/egress gateway component can also have multiple instances). Another problem I want to point out is that when the manifests are applied, there's a dependency graph that's imposed based on what is currently ComponentName. We need some concept that will continue to work with the current dependency graph, which can't have dynamically named components in it. I've been treating ComponentName as a sort of "component category" concept with multiple instances differentiated by ResourceName. But this can definitely use some thought to express better and I'm happy to have a chat about it if you haven't lost your appetite for this change :-)